src/test/java/com/metamx/common/io/smoosh/SmooshedFileMapperTest.java

```diff
@@ -132,7 +132,7 @@ public void testBehaviorWhenReportedSizesSmall() throws Exception
       Assert.assertTrue(exceptionThrown);
       File[] files = baseDir.listFiles();
       Assert.assertEquals(1, files.length);
-      Assert.assertEquals(0, files[0].length());
+      Assert.assertEquals(4, files[0].length());
```

Comment:
it was buffered stream before but is not, making that difference. I've implemented buffered channel and it's restored as before.